### PR TITLE
Feature/en 531 docs site table component

### PR DESCRIPTION
**Pull request by m7kvqbe1**
## Related Jira issue:

https://rn-nelson.atlassian.net/browse/EN-531

## Overview

Implement the docs-site sortable responsive table component.

## Reason

>The docsite uses a standard table implementation as per the linked design ticket. However, there is no provision for optimising the display of a table on a mobile device in the ticket.

## Work carried out

- [x] Implement docs-site specific sortable responsive table component
- [x] Add automated tests
- [x] Perform UAT down to Chrome 38 (and various form factors)

## Screenshot

Screenshot 2019-06-13 at 17 57 54
Screenshot 2019-06-13 at 10 57 14

## Developer notes

Note that because of the explicit responsive requirements I've implemented this as a docs-site component. 

We need to decide how this is handled in the component library implementation (if at all). I'm mindful of concerns previously raised about doc-site design choices influencing component-library implementations.

### CODE REVIEW

File: packages/docs-site/src/components/presenters/data-table/index.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  const sortByColumn = useCallback(
+    e => {
+      const column = e.target.getAttribute('data-column')
+      setTableData(sortBy(data, column))
+    },
+    [tableData]
+  )
+
+  return (
+    <div className="data-table">
```
Comment (m7kvqbe1) [inline packages/docs-site/src/components/presenters/data-table/index.js]: I wish I didn't need to do this - I had to wrap the table to sort out some styling issues - the caption element doesn't play nice with `box-shadow` on a table.

Comment (m7kvqbe1) [reply packages/docs-site/src/components/presenters/data-table/index.js]: Screenshot 2019-06-13 at 17 57 54

Comment (m7kvqbe1) [reply packages/docs-site/src/components/presenters/data-table/index.js]: @hxltrhuxze See if you can get `border-radius` to work on the bottom corners - I struggled with this. It seems to be an issue with `border-collapse: collapse;`.

**Review by antictor — APPROVED** [review]

File: packages/docs-site/src/components/presenters/data-table/data-table.scss [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+      th:first-child, td:first-child {
+        border-left: 1px solid $border-color;
+      }
+
+      // Target the last table row and set the 
+      // border radius on the first and last element
+      &:last-child {
+        td:first-child {
+          border-bottom-left-radius: 6px;
```
Comment (m7kvqbe1) [inline packages/docs-site/src/components/presenters/data-table/data-table.scss]: @hxltrhuxze 

Should this be 4px and use `$border-radius`?

**Review by 09txzS1qs — COMMENTED** [review]
@m7kvqbe1 are you still waiting for a response on the 4px question? If wrapping issue is resolved and it's just minor styling tweaks then happy to approve.

File: packages/docs-site/src/components/presenters/data-table/data-table.scss (hunk 2 above)
Comment (hxltrhuxze) [reply packages/docs-site/src/components/presenters/data-table/data-table.scss]: Good spot. You're totally right
```suggestion
          border-bottom-left-radius: $border-radius;
```

File: packages/docs-site/src/components/presenters/data-table/data-table.scss [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+        }
+        td:last-child {
+          border-bottom-right-radius: 6px;
```
Comment (hxltrhuxze) [inline packages/docs-site/src/components/presenters/data-table/data-table.scss]:
```suggestion
          border-bottom-right-radius: $border-radius;
```

**Pull request merged**